rmw_cyclonedds_cpp/src/serdata.cpp

```diff
@@ -504,7 +504,7 @@ void serdata_rmw::resize(size_t requested_size)
 
   /* FIXME: CDR padding in DDSI makes me do this to avoid reading beyond the bounds
   when copying data to network.  Should fix Cyclone to handle that more elegantly.  */
-  size_t n_pad_bytes = (-requested_size) % 4;
+  size_t n_pad_bytes = 4 - (requested_size % 4);  // Same as (-x) % 4, without negating unsigned
```

Comment:
Thanks for fixing helping out and addressing the warnings — but beware: this is not equivalent to the original. If you assume ``requested_size`` is 0, then the original trivially evaluates to 0 and the proposed to 4. The damage will be very minor because — I suppose all — DDS implementations will ignore the additional bytes at the end of the serialised data, but it is a waste to add 4 bytes to any sample that happens to have a serialised size that is a multiple of 4 bytes.

 There are various solutions. I think I'd go for defining ``aligned_size = (requested_size + 3) & ~3``, and then replacing ``requested_size+n_pad_bytes`` to ``aligned_size`` and the bare ``n_pad_bytes`` in the ``memset`` with ``aligned_size-requested_size``.

Other reasonable choices would be to use ``(0-requested_size)%4`` or ``(4-(requested_size%4))%4``, or …

I guess one would need to verify which MS's compiler finds acceptable …

Reply:
Updated to try one of the versions
Windows Packaging: [![Build Status](https://ci.ros2.org/buildStatus/icon?job=ci_packaging_windows&build=141)](https://ci.ros2.org/job/ci_packaging_windows/141/)